docs/tutorials/dwithin-join.rst

```diff
@@ -0,0 +1,188 @@
+GeoMesa Spark: Aggregating Data
+===============================
+
+This tutorial will show you how to:
+
+1. Use GeoMesa with `Apache Spark <http://spark.apache.org/>`__ in Scala.
+2. Create and use DataFrames with our geospatial User Defined Functions.
+3. Calculate aggregate statistics based on a threshold distance.
+4. Create a new simple feature type to represent this aggregation.
+5. Write the result back to a data store.
+
+Background
+----------
+
+`NYCTaxi <https://databank.illinois.edu/datasets/IDB-9610843>`__ is  taxi activity data
+published by the University of Illinois from Freedom of Information Law requests to NYC Taxi and Limo Commission.
+
+`GeoNames <http://www.geonames.org>`__ is a geographical database containing over
+10 million geographical names and over 9 million unique features.
+
+Suppose we wanted to answer the questions: "Do taxi pickups centralize near certain points of interest?",
+"Are people more likely to request a pickup or be dropped off at points of interest?". To find out, we would need to
+combine these two data sets and aggregate statistics over the result.
+
+Since GeoNames is a data set of points, and NYCTaxi provides only the pickup and drop-off points of a trip, it is highly
+unlikely that the trip would start or end exactly on the labeled point of interest. So we can't naively join
+the data sets based on where the points are equal. Instead, we will need to join where the points are within some tolerable
+distance of each other. This is henceforth referred to as a D-within join.
+
+Prerequisites
+-------------
+
+For this tutorial, we will assume that your have already ingested the two data sets into the data store of your choosing.
+If you haven't, you can follow one of these tutorials :doc:`/tutorials/geomesa-examples-gdelt`. The converters for these
```

Comment:
in addition to the caveats mentioned above, it would be useful to link to our converters page for these two data types (i.e. http://www.geomesa.org/documentation/user/convert/premade/nyctaxi.html, etc)